github/codeowners.py

```diff
@@ -12,21 +12,51 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import collections.abc
+import dataclasses
 from pathlib import Path
 import logging
 import typing
 
-from github3 import GitHub
+from github3 import GitHub, GitHubEnterprise
 from github3.exceptions import NotFoundError
+import github3.exceptions
+import github3.orgs
 import github3.repos.repo
+import github3.search.user
+import github3.users
 
-from ci.util import existing_dir, existing_file, not_none
+from ci.util import existing_dir, existing_file
 import ci.log
 
 logger = logging.getLogger(__name__)
 ci.log.configure_default_logging()
 
 
+@dataclasses.dataclass(frozen=True)
+class Codeowner:
+    value: str
+
+    def __str__(self):
+        return self.value
+
+
+@dataclasses.dataclass(frozen=True)
+class Username(Codeowner):
+    pass
+
+
+@dataclasses.dataclass(frozen=True)
+class EmailAddress(Codeowner):
+    pass
+
+
+@dataclasses.dataclass(frozen=True)
+class Teamname(Codeowner):
+    # must be in format org/team
```

Comment:
well... validate in initialiser? also, the (raw) format also demands a leading `@` character.

Reply:
also, add properties for parsing, like so:
```
@property
def org_name(self): <return org, parsed from raw teamname>

@property
def team_name(self): <return non-qualified teamname, parsed from raw teamname>
```

considering there seem to be two flavours of teamnames ("qualified", i.e. including org-prefix and "unqualified" (name only)), the name seems to be chose a bit unfortunately..